### PR TITLE
EC2MetadataUtils: add marketplaceProductCodes inside InstanceInfo's POJO

### PR DESCRIPTION
**Pull request by notdryft**
EC2MetadataUtils' InstanceInfo is missing the product codes field that is filled when running an instance through AWS Marketplace.

The field is named `marketplaceProductCodes`, is a sequence of String and can be null. See sample instance identify document on a non marketplace instance:
```json
{
  "devpayProductCodes" : null,
  "marketplaceProductCodes" : null,
  "accountId" : "XXX",
  "availabilityZone" : "eu-west-3c",
  "ramdiskId" : null,
  "kernelId" : null,
  "pendingTime" : "2019-04-15T15:51:52Z",
  "architecture" : "x86_64",
  "privateIp" : "X.X.X.X",
  "version" : "2017-09-30",
  "region" : "eu-west-3",
  "billingProducts" : null,
  "instanceId" : "i-XXXX",
  "imageId" : "ami-XXX",
  "instanceType" : "t2.medium"
}
```

### CODE REVIEW

File: core/regions/src/test/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtilsServer.java [hunk 1]
```diff
@@ -213,7 +213,8 @@ protected String constructInstanceInfo() {
                + "\"version\":\"2010-08-31\","
                + "\"availabilityZone\":\"us-east-1b\","
                + "\"privateIp\":\"10.201.215.38\","
-               + "\"devpayProductCodes\":[\"bar\"]"
+               + "\"devpayProductCodes\":[\"bar\"],"
+               + "\"marketplaceProductCodes\":null"
```
Comment (zoewangg) [inline core/regions/src/test/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtilsServer.java]: Can we make it  a non-null value and assert that value instead?

Comment (notdryft) [reply core/regions/src/test/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtilsServer.java]: Sure

**Comment by notdryft** [comment]
Updated with non null test, squashed commit and new entry to change log!

File: core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtils.java [hunk 2]
```diff
@@ -574,6 +578,10 @@ public String getPrivateIp() {
         public String[] getDevpayProductCodes() {
             return devpayProductCodes == null ? null : devpayProductCodes.clone();
         }
+
+        public String[] getMarketplaceProductCodes() {
+            return marketplaceProductCodes == null ? null : devpayProductCodes.clone();
```
Comment (zoewangg) [inline core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtils.java]: `marketplaceProductCodes.clone()`?

Comment (notdryft) [reply core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtils.java]: Way too hasty on this one... :<

Comment (notdryft) [reply core/regions/src/main/java/software/amazon/awssdk/regions/internal/util/EC2MetadataUtils.java]: Fixed

**Comment by notdryft** [comment]
Updated

**Pull request merged**